unionFindWithWeight.js

```diff
@@ -3,21 +3,48 @@
 class unionFindWithWeight {
 	constructor() {
 		this._elements = [];
+		this._weigthOfComponents = {};
+	}
+
+	getMainRootElement(a) {
+		if (isNaN(this._elements[a])) {
+			return -1;
+		} else if (this._elements[a] === a) {
+			return this._elements[a];
+		} else {
+			return this.getMainRootElement(this._elements[a]);
+		}
 	}
```

Comment:
good to make it with recursion
but in js it bad practices (you get a stack overflow)

it's ok it leave it as is